src/main/java/com/archisacadeny/course/CourseService.java

```diff
@@ -157,4 +157,13 @@ public Map<String,Object> generateCourseReport(int courseId) {
         return values;
     }
 
+    public List<Course> listCoursesOrderedByStudentAverageGrade() {
+        List<Course> courses = courseRepository.listCoursesOrderedByStudentAverageGrade();
+
+        for (Course course : courses) {
+            System.out.println(course);
```

Comment:
this is going to print the course reference number. you should print course details at least you can print the course name

Reply:
Hocam I have an overriteen toString method in Course class that does it, should i manually print it or its good ?